README.md

```diff
@@ -4,16 +4,30 @@
 
 ## Description
 
-Full text search engines like Apache Lucene are very powerful technologies to add efficient free
-text search capabilities to applications. However, Lucene suffers several mismatches when dealing
-with object domain models. Amongst other things indexes have to be kept up to date and mismatches
-between index structure and domain model as well as query mismatches have to be avoided.
+Full text search for Java objects
 
-Hibernate Search addresses these shortcomings - it indexes your domain model with the help of a few
-annotations, takes care of database/index synchronization and brings back regular managed objects
-from free text queries.
+This project provides synchronization between entities managed by Hibernate ORM and full-text
+indexing services like Apache Lucene and Elasticsearch.
 
-Hibernate Search is using [Apache Lucene](http://lucene.apache.org/) under the cover.
+It will automatically apply changes to indexes, which is tedious and error prone coding work,
+while leaving you full control on the query aspects. The development community constantly
+researches and refines performance improvements to apply the writes at optimal performance.
+
+Mapping your objects to the indexes is declarative, using a combination of Hibernate Search
+specific annotations and the knowledge it can gather from your existing Hibernate/JPA
+mapping.
+
+Queries can be defined by any combination of:
+ - "native" Apache Lucene queries
+ - writing "native" Elasticsearch queries in Json format
+ - using our DSL which abstracts the previous two generating optimal backend specific queries
+
+Query results can use projections to be loaded directly from the index, or can materialize
+fully managed Hibernate entities loaded from the database within the current transactional scope.
+
+Hibernate Search is using [Apache Lucene](http://lucene.apache.org/) under the cover; this
+can be used directly (running embedded in the same JVM) or remotely provided by Elasticsearch
+over its REST API.
```

Comment:
I need the first paragraph to be short. Here I'm indeed repeating the concept, as a closing point.. but  the real message is to make people aware that even if they opt for using ES it's still Lucene under the covers.